src/lib/storage/table.cpp

```diff
@@ -18,56 +18,80 @@
 namespace opossum {
 
 Table::Table(const ChunkOffset target_chunk_size) {
-  // Implementation goes here
+  _target_chunk_size = target_chunk_size;
+  _chunks.emplace_back(Chunk());
 }
 
 void Table::add_column(const std::string& name, const std::string& type) {
-  // Implementation goes here
+  DebugAssert(_chunks.size() == 1, "Invalid amount of chunks");
+  DebugAssert(row_count() == 0, "Too many rows");
+
+  _add_segment(type);
+
+  _column_names.emplace_back(name);
+  _column_types.emplace_back(type);
 }
 
-void Table::append(const std::vector<AllTypeVariant>& values) {
-  // Implementation goes here
+void Table::_add_segment(const std::string& type) {
+  resolve_data_type(type, [&](const auto data_type_t) {
+    using ColumnDataType = typename decltype(data_type_t)::type;
+    const auto value_segment = std::make_shared<ValueSegment<ColumnDataType>>();
+    _chunks.back().add_segment(value_segment);
+  });
 }
 
-ColumnCount Table::column_count() const {
-  // Implementation goes here
-  return ColumnCount{0};
+void Table::_create_new_chunk() {
+  _chunks.emplace_back(Chunk());
+  for (size_t column_id = 0; column_id < _column_names.size(); ++column_id) {
```

Comment:
```suggestion
  for (auto column_id = 0, columns_size = _column_names.size(); column_id < columns_size; ++column_id) {
```
To avoid assuming the vector size to be size_t, maybe use auto.
Furthermore, the loop condition is evaluated every time, this could be fixed by evaluating it once before the loop starts, see also https://llvm.org/docs/CodingStandards.html#don-t-evaluate-end-every-time-through-a-loop